tests/lib/rules/no-invalid-attribute-name.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * @author *****your name*****
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/no-invalid-attribute-name')
+
+const tester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module'
+  }
+})
+
+tester.run('no-invalid-attribute-name', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: '<template><p foo /></template>'
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p foo="bar" /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p foo-bar /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p _foo-bar /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p :foo-bar /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p foo.bar /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p quux-.9 /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><MyComponent 0abc="foo" /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><MyComponent :0abc="foo" /></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><a :href="url"> ... </a></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><div v-bind:class="{ active: isActive }"></div></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><p v-if="seen">Now you see me</p></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<a v-on:[eventName]="doSomething"> ... </a>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<form v-on:submit.prevent="onSubmit"> ... </form>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<a @[event]="doSomething"> ... </a>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><div v-bind:href="..."></div></template>`
+    },
+    {
+      filename: 'test.vue',
+      code: `<template><div :href="..."></div></template>`
+    }
```

Comment:
These cases are already by the test cases above (`<a :href="url">` and `<div v-bind:class="{ active: isActive }">`). But I'd add these test cases instead:

```suggestion
    {
      filename: 'test.vue',
      code: `<template><div v-bind="..."></div></template>`
    },
    {
      filename: 'test.vue',
      code: `<template><div v-0abc="..."></div></template>`
    }
```